jenkins/data-prepper/Jenkinsfile

```diff
@@ -0,0 +1,55 @@
+lib = library(identifier: 'jenkins@20211123', retriever: legacySCM(scm))
+
+pipeline {
+    agent {
+        docker {
+            label 'Jenkins-Agent-al2-x64-c54xlarge-Docker-Host'
+            image 'opensearchstaging/ci-runner:ubuntu2004-x64-docker-buildx0.6.3-qemu5.0-awscli1.22-jdk14'
+            alwaysPull true
+        }
+    }
+    parameters {
+        string(name: 'VERSION', description: 'The version of Data Prepper', trim: true)
+        string(name: 'BRANCH', description: 'Can be: <branchName>, <commitId>, refs/tags/<tagName>', trim: true)
+    }
+    stages {
+        stage('Build Data Prepper') {
+            steps {
+                script {
+                    checkout changelog: false, poll: false, scm:
+                        [$class: 'GitSCM', branches: [[name: BRANCH]], userRemoteConfigs: [[url: 'https://github.com/opensearch-project/data-prepper.git']]]
+                    sh './gradlew clean :release:archives:linux:linuxTar -Prelease'
+                }
+            }
+        }
+        stage('Sign Artifacts') {
+            steps {
+                script {
+                    signArtifacts(
+                        artifactPath: "$WORKSPACE/release/archives/linux/build/distributions/",
+                        signatureType: '.sig',
+                        distributionPlatform: 'linux'
+                    )
+                }
+            }
+        }
+        stage('Upload Artifacts to Staging Distribution Bucket') {
```

Comment:
Future step, it might make sense to tie the results that are uploaded to a slack notification [example](https://github.com/opensearch-project/opensearch-build/blob/84d8da0b042f6cb87569f2163776140eb733ef66/jenkins/opensearch/distribution-build.jenkinsfile#L121).

Reply:
> Future step, it might make sense to tie the results that are uploaded to a slack notification [example](https://github.com/opensearch-project/opensearch-build/blob/84d8da0b042f6cb87569f2163776140eb733ef66/jenkins/opensearch/distribution-build.jenkinsfile#L121).

@dlvenable - Maybe add this feature as a part of #1439 issue?